src/cpp/garbage-collector.h

```diff
@@ -20,11 +20,13 @@ class GC {
         if (enabled_ && count() >= step_ * lastCleanup_)
             collect();
 
-        std::unique_ptr<ViewType> object(new ViewType(std::forward<Args>(args)...));
+        std::unique_lock<std::mutex> g(lock_);
+        std::unique_ptr<ViewType> object(new ViewType);
```

Comment:
Better use `std::make_unique<ViewType>`

Reply:
nope, cause  constructor of ViewType is private. There was a `new` instead of `make_unique` exactly because of that. And that was your idea))